plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/AbstractFilesTable.java

```diff
@@ -89,22 +91,47 @@ protected Table getIcebergTable()
         return icebergTable;
     }
 
+    private ConnectorTableMetadata tableMetadata(SchemaTableName tableName, TypeManager typeManager)
+    {
+        return new ConnectorTableMetadata(requireNonNull(tableName, "tableName is null"),
+                ImmutableList.<ColumnMetadata>builder()
+                        .add(new ColumnMetadata("content", INTEGER))
+                        .add(new ColumnMetadata("file_path", VARCHAR))
+                        .add(new ColumnMetadata("file_format", VARCHAR))
+                        .add(new ColumnMetadata("record_count", BIGINT))
+                        .add(new ColumnMetadata("file_size_in_bytes", BIGINT))
+                        .add(new ColumnMetadata("column_sizes", typeManager.getType(mapType(INTEGER.getTypeSignature(), BIGINT.getTypeSignature()))))
+                        .add(new ColumnMetadata("value_counts", typeManager.getType(mapType(INTEGER.getTypeSignature(), BIGINT.getTypeSignature()))))
+                        .add(new ColumnMetadata("null_value_counts", typeManager.getType(mapType(INTEGER.getTypeSignature(), BIGINT.getTypeSignature()))))
+                        .add(new ColumnMetadata("nan_value_counts", typeManager.getType(mapType(INTEGER.getTypeSignature(), BIGINT.getTypeSignature()))))
+                        .add(new ColumnMetadata("lower_bounds", typeManager.getType(mapType(INTEGER.getTypeSignature(), VARCHAR.getTypeSignature()))))
+                        .add(new ColumnMetadata("upper_bounds", typeManager.getType(mapType(INTEGER.getTypeSignature(), VARCHAR.getTypeSignature()))))
+                        .add(new ColumnMetadata("key_metadata", VARBINARY))
+                        .add(new ColumnMetadata("split_offsets", new ArrayType(BIGINT)))
+                        .add(new ColumnMetadata("equality_ids", new ArrayType(INTEGER)))
+                        .build());
+    }
+
     private List<Page> buildPages()
     {
         PageListBuilder pagesBuilder = PageListBuilder.forTable(tableMetadata);
         Map<Integer, Type> idToTypeMapping = getIcebergIdToTypeMapping(icebergTable.schema());
         TableScan tableScan = buildTableScan();
 
         tableScan.planFiles().forEach(fileScanTask -> {
-            addRow(pagesBuilder, idToTypeMapping, fileScanTask.file());
+            DataTask dataTask = (DataTask) fileScanTask;
+            dataTask.rows().forEach(dataTaskRow -> addRow(pagesBuilder, idToTypeMapping, dataTaskRow));
         });
 
         return pagesBuilder.build();
     }
 
-    private void addRow(PageListBuilder pagesBuilder, Map<Integer, Type> idToTypeMapping, DataFile dataFile)
+    private static void addRow(PageListBuilder pagesBuilder, Map<Integer, Type> idToTypeMapping, StructLike dataTaskRow)
```

Comment:
take into account that you could receive delete files here as well. Pass `ContentFile` 

```
private static <F> void addRow(PageListBuilder pagesBuilder, Map<Integer, Type> idToTypeMapping, ContentFile contentFile)
```

Do also check in `TestIcebergSystemTables` whether the delete files are being taken into account.

Reply:
@findinpath Actually I looked into the `ContentFile` but as Iceberg supported deletes and has added `DeleteFilesTable` https://github.com/apache/iceberg/blob/master/core/src/main/java/org/apache/iceberg/DeleteFilesTable.java#L24
So, IMO Tino's FilesTable represents the DataFiles only. 
After this, I'm planning to add `DeleteFilesTable` to Trino as well.

Reply:
@findinpath is the comment can be resolved? thanks